lib/chewy/strategy/atomic.rb

```diff
@@ -20,12 +20,20 @@ def update type, objects, options = {}
         ActiveSupport::Deprecation.warn("`urgent: true` option is deprecated and is not effective inside `:atomic` strategy, use `Chewy.strategy(:urgent)` strategy instead") if options.key?(:urgent)
 
         @stash[type] ||= []
-        @stash[type] |= type.adapter.identify(objects)
+        @stash[type] |= _identify_objects type, objects
       end
 
       def leave
         @stash.all? { |type, ids| type.import!(ids) }
       end
+
+      private
+
+      def _identify_objects type, objects
+        identifiers_from_evaluator = type.root_object.compose_id(objects) if type.root_object
```

Comment:
I was a bit confused about that point.

1. Under what circumstances might objects be a collection?

2. The method name `compose_id` means to build the id for an object, or does compose refer to some other form of software engineering composition technique?